src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -105,7 +105,7 @@ public function add(array $headers)
      * @param mixed  $default The default value
      * @param bool   $first   Whether to return the first value or all header values
      *
-     * @return string|array The first header value if $first is true, an array of values otherwise
+     * @return string|array|mixed The first header value or default value if $first is true, an array of values otherwise
```

Comment:
Shouldn't we change the type for the `$default` argument to `string|array` instead?

Reply:
if we allow that implicitly, yes please. Might even consider `string|string[]` to be really clear (updating `set()` while at it).

But im not sure technically we need to trigger a deprecation for non-string values passed today, or cast those tomorrow.

Otherwise :+1: for adding `|mixed` to help clarify.

Reply:
phpdocs are just docs, there is no BC promise on them. Code only matters.